File: Python DSA/HashMap/contains_duplicate.py
"""
CONTAINS DUPLICATE - Easy
Return true if any value appears at least twice in array.

Logical Thinking:
1. Use set to track seen values
2. If value already in set, return True
3. Otherwise add to set

Logical Thinking:
def contains_duplicate(nums):
    seen = set()
    for num in nums:
        if num in seen:
            return True
        seen.add(num)
    return False
"""

# TODO: Implement solution


# Test cases
if __name__ == "__main__":
    print(f"Testing Contains Duplicate...")
    print("Add your test cases here")

File: DSA/Array/Java/ContainsDuplicate.java
// LeetCode 217: Contains Duplicate
// Given an integer array nums, return true if any value appears at least twice in the array.

import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        
        for (int num : nums) {
            if (seen.contains(num)) {
                return true;
            }
            seen.add(num);
        }
        
        return false;
    }
    
    public static void main(String[] args) {
        ContainsDuplicate solution = new ContainsDuplicate();
        int[] nums1 = {1, 2, 3, 1};
        int[] nums2 = {1, 2, 3, 4};
        
        System.out.println("Contains duplicate: " + solution.containsDuplicate(nums1));
        System.out.println("Contains duplicate: " + solution.containsDuplicate(nums2));
    }
}